packages/synapse-react-client/tsconfig.json

```diff
@@ -1,15 +1,52 @@
 {
   "extends": "../../shared/tsconfig.base.json",
   "compilerOptions": {
+    "composite": true,
     "rootDir": "./src",
-    "outDir": "./dist/types",
+    "outDir": "../../node_modules/.cache/.tsc_cache/synapse-react-client",
     "module": "esnext",
     "target": "esnext",
     "moduleResolution": "bundler",
     "allowJs": false,
-    "noEmit": true,
+    "noEmit": false,
     "isolatedModules": true,
-    "noFallthroughCasesInSwitch": true
+    "noFallthroughCasesInSwitch": true,
+    "paths": {
+      // HMR will work if we uncomment these, but doing so causes various build/typecheck errors
+      //
+      // In the markdown-it packages, the TypeScript compiler had an issue with the output type declaration file being
+      // generated by tsup and not TypeScript. We could probably replace tsup with the TypeScript compiler, but that
+      // might cause breaking changes that must be handled in SWC.
+      //
+      // In the synapse-client and synapse-types packages cause failures when parsed by the vite-plugin-dts plugin
+      // that we use to generate a single type declaration file in synapse-react-client
+
```

Comment:
Incremental type-checking and HMR in synapse-react-client won't work for these dependencies. We can probably deal with them, but I didn't want to deal with it right now.

IMO not a big deal, since these dependencies are pretty small and easy to rebuild. The big advantage to this PR is getting HMR for synapse-react-client components in Vite apps.